connectors/byoc.py

```diff
@@ -630,26 +566,38 @@ async def _sync_done(self, job, status, result, exception=None):
             await job.fail(exception, ingestion_stats=ingestion_stats)
         elif status == JobStatus.SUSPENDED:
             await job.suspend(ingestion_stats=ingestion_stats)
-            logger.info(f"Sync suspended, Job id: {job.id}")
         else:
             await job.done(ingestion_stats=ingestion_stats)
 
         job = await job.reload()
         await self.sync_done(job)
         logger.info(
             f"[{job_id}] Sync done: {doc_updated + doc_created} indexed, {doc_deleted} "
-            f" deleted. ({int(time.time() - self._start_time)} seconds)"
+            f" deleted. ({int(time.time() - start_time)} seconds)"
         )
 
     async def sync_done(self, job):
-        self.doc_source["last_sync_status"] = job.status.value
-        self.doc_source["last_sync_error"] = job.error
-        self.doc_source["error"] = job.error
-        self.status = (
-            Status.ERROR if job.status == JobStatus.ERROR else Status.CONNECTED
+        job_status = JobStatus.ERROR if job is None else job.status
+        job_error = JOB_NOT_FOUND_ERROR if job is None else job.error
+        if job_error is None and job_status == JobStatus.ERROR:
+            job_error = UNKNOWN_ERROR
+        connector_status = (
+            Status.ERROR if job_status == JobStatus.ERROR else Status.CONNECTED
         )
-        self.doc_source["last_synced"] = iso_utc()
-        await self.sync_doc()
+
+        doc = {
+            "last_sync_status": job_status.value,
+            "last_synced": iso_utc(),
+            "last_sync_error": job_error,
+            "status": connector_status.value,
+            "error": job_error,
+        }
+
+        if job is not None and job.terminated:
+            doc["last_indexed_document_count"] = job.indexed_document_count
+            doc["last_deleted_document_count"] = job.deleted_document_count
+
+        await self.index.update(doc_id=self.id, doc=doc)
 
     async def prepare_docs(self, data_provider, pipeline=None, filtering=None):
```

Comment:
`prepare_docs` will eventually be removed once job execution is moved out of `Connector` class in a separate PR.

Reply:
can you add a `TODO`

Reply:
Done in https://github.com/elastic/connectors-python/pull/474/commits/1224df3fe0836512deffca22102671570b8ca9a7